src/context/todoContext/reducer.ts

```diff
@@ -0,0 +1,45 @@
+import { LSHelper } from 'utils';
+import { CREATE, DELETE, LOAD } from './actionTypes';
+import { IState, Action, CreateTodoDto} from 'types/context';
+import { ITodo, TPriority, TStatus } from 'types/todo';
+import { TODOS } from 'utils/contants';
+
+export default function reducer(state: IState, action: Action): IState {
+  const { type, payload } = action;
+  switch (type) {
+    case LOAD:
+      return loadTodos();
+    case CREATE:
+      const newTodo = createTodo(state.nextId, payload)
+      return {
+        todos : state.todos.concat(newTodo),
+        nextId: newTodo.id + 1,
+      }
+    case DELETE:
+      return { ...state };
+    default:
+      return { ...state };
+  }
+}
+
+const loadTodos = (): IState => {
+  const todos = LSHelper.getItem(TODOS) || [];
+  const nextId = todos.length ? Math.max(...todos.map((todo: ITodo) => todo.id)) + 1 : 0
+  return { todos,  nextId: nextId};
+}
+
+const createTodo = (nextId: number, createTodoDto: CreateTodoDto): ITodo => {
```

Comment:
createTodoDto 이름 관련한 제 사소한 제안을 말씀드립니다!
Dto가 Data transfer object라는 뜻의 전문용어라서 좋은 것 같습니다! 덕분에 하나 배웠습니다. 👍👍
다만, 제가 아직 백엔드 용어에 익숙하지 않다보니 처음에 조금 직관적으로 이해하기 어려웠어요😭

그리고 제 생각에는 객체를 지칭하는 명칭인데 "create"라는 동사가 들어가서 조금 헷갈렸던 것 같습니다!


그래서 `CreateTodoDto`를 `TodoFormInput`이나 `TodoFormValue`로  바꾸는 건 어떨지 살짝쿵 제안드려봅니다!
(제가 추후 `TodoForm.tsx` 같은 컴포넌트를 만든다고 했을 때 조금 더 연결성도 있을 것 같아요!)

정말 정말 가벼운 제안이니 반영 안하셔도 괜찮습니다 😆👍👍👍👍


```suggestion
const createTodo = (nextId: number, todoFormValue: TodoFormValue): ITodo => {
```

Reply:
다시 생각해보니 `CreateTodoDto` 가 어울리지는 않을거같네요. (왜냐면 백엔드에서 사용하는 개념이니깐요!)
다만 말씀해주신 `TodoFormValue` 는 목적이 명확하지 않은거같아요.
왜냐면 `form` 이라는게 업데이트에서 사용될 수도, 생성에 사용될 수도 있으니깐요.
이 부분은 다른 분들이랑 다시 이야기를 나눠보는게 어떨까요? 👍 
(사실 타입 이름을 어떻게 지어야할지 감이 잘 안와요ㅜㅜ)

Reply:
넵넵! 저도 `TodoFormValue` 라는 이름에 대해서 확신이 없었습니다 😭
일단 그냥 두시거나 얘기 나눠보고 수정해보면 좋을 것 같습니다~~! 👍👍👍👍